roadmap/parachains.md

```diff
@@ -0,0 +1,184 @@
+
+Category: Node, Networking
+
+For each potential, considered parachain block, perform and erasure-coding of the PoV and outgoing messages of the block. Call the number of validators on the relay chain for the Relay-chain block this parachain block is being considered for inclusion in `n`. Erasure-code into `n` pieces, where any `f + 1` can recover (`f` being the maximum number of tolerated faulty nodes = ~ `n / 3`). The `i'th` validator stores the `i'th` piece of the coding and provides it to any who ask.
+
+#### *PoV block fetching*
+
+Category: Networking
+
+A black-box networking component for validators or fishermen on a parachain to obtain the PoV block referenced by hash in an attestation, for the purpose of validating. When fetching "current" PoV blocks (close to the head of the chain, or relating to the block currently being built), this should be fast. When fetching "old" PoV blocks, it should be possible and fall back on recovering from the availability erasure-coding.
+
+#### *Parathread Auction Voting*
+
+Category: Node, Networking
+
+How and when collators are configured to cast votes in parathread auctions.
+
+#### *Collation Loop*
+
+Category: Node, Networking
+
+The main event loop of a collator node:
+ 1. new relay chain block B
+ 2. sync new parachain head w.r.t. B
+ 3. build new child of B
+ 4. submit to validators
+
+---
+### Cross-chain Messaging
+
+https://hackmd.io/ILoQltEISP697oMYe4HbrA?view
+https://github.com/paritytech/polkadot/issues/597
+
+The biggest sub-project of the parachains roadmap - how messages are sent between parachains.
+
+TODO: break down further.
+
+---
+### Fishing/Slashing
+
+#### *Validity/Availability Report Handler*
+
+Category: Runtime
+
+In Polkadot, a bad parachain group can force inclusion of an invalid or unavailable parachain block. It is the job of fishermen to detect those blocks and report them to the runtime. This item is about the report handler
+
+The W3F-research writeup on availability/validity provides a high-level view of the dispute resolution process: [Availability and Validity — Research at W3F](https://research.web3.foundation/en/latest/polkadot/Availability_and_Validity.html)
+
+One of the main behaviors that is unimplemented and needs to be is the _rollback_ that occurs when the dispute resolution process concludes that an error has been made. When we mark a parachain block as having been invalid or unavailable, we need to roll back all parachains to a point from just before this state.  We would also need to roll back relay chain state, because there may have been messages from a parachain to a relay chain that now need to be rolled back.  The easiest thing to do would be to side-step that by putting a delay on upwards messages, but this would impact the UX of parachain participation in slot auctions, council votes, etc. considerably. Assuming we can't side-step this, we will have to find a way to roll back selected state of the relay chain.
```

Comment:
```suggestion
One of the main behaviors that is unimplemented and needs to be is the _rollback_ that occurs when the dispute resolution process concludes that an error has been made. When we mark a parachain block as having been invalid or unavailable, we need to roll back all parachains to a point from just before this state.  We would also need to roll back relay chain state, because there may have been messages from a parachain to a relay chain that now need to be rolled back. The easiest thing to do would be to side-step that by putting a delay on upwards messages, but this would impact the UX of parachain participation in slot auctions, council votes, etc. considerably. Assuming we can't side-step this, we will have to find a way to roll back selected state of the relay chain.
```

Reply:
what changed here? hard for me to see